src/test/java/javax/jmdns/test/JmDNSTest.java

```diff
@@ -521,4 +528,30 @@ public void testListMyServiceWithoutLowerCase() throws IOException, InterruptedE
             if (registry != null) registry.close();
         }
     }
+
+    @Test
+    public void testRegisterServiceWithDifferentDefaultTLL() throws IOException, InterruptedException {
+        System.out.println("Unit Test: testRegisterServiceWithDifferentDefaultTLL()");
+        JmDNS registry = null;
+        int defaultTTL = 30;
+        try {
+            DNSStateTask.setDefaultTTL(defaultTTL);
+            registry = JmDNS.create();
+            registry.registerService(service);
+            // Needed - otherwise DNSQuestion.addAnswersForServiceInfo is not called
+            ServiceInfo[] services = registry.list(service.getType());
```

Comment:
Right, but you can remove the `ServiceInfo[] services = ` to avoid "unused" warnings in the IDE.